### PR TITLE
Some improvements in the external commands tests

### LINKED ISSUE #477: External commands not logged
When the receiver is configured for a `direct_routing` to its scheduler, the received external commands are not logged ...

### LINKED ISSUE #484: Specific Alignak external commands
Alignak has some specific external commands:
 - del_host_dependency,
 - add_simple_host_dependency,
 - add_simple_poller

Are those commands really useful? Should they be maintained in the framework?

### LINKED ISSUE #486: External command CHANGE_SVC_MODATTR should be a toggle
This command should be a toggle on/off for each concerned item, whereas it always set On and never Off !

### LINKED ISSUE #487: Contacts objects do not have customs property ...
In a unit test`module:
```
        contact = self._scheduler.contacts.find_by_name("test_contact")
        self.assertIsNotNone(contact)
        self.assertEqual(contact.contact_name, "test_contact")
        self.assertIsNotNone(contact.customs)
        self.assertEqual(contact.customs['_CUSTOM_CONTACT_VAR1'], 10)
        self.assertEqual(contact.customs['_CUSTOM_CONTACT_VAR2'], 'text')
```

But `contact.customs` is None !

### LINKED ISSUE #489: Contacts downtimes raise exception
When deleting a contact downtime with an external command, an error is raised because the Contact object do not have a `broks` property

### PR DESCRIPTION
**Pull request by mohierf**
Improve external commands tests:
- host/service/contact update attributes commands
- hosts / services comments
- hosts / services downtimes
- not implemented commands
- hostgroups commands
- hosts commands
- hosts global commands
- global commands
- contacts downtimes (but test skipped because of a bug in the contacts downtimes)
- monitoring logs
- servicegroups commands
- remove enable_failure_prediction (not implemented in Alignak)

Closes #477: external commands not logged
Closes #487: missing customs in contacts
Closes #486: toggle for CHANGE_SVC_MODATTR  command
Closes #484: remove Alignak specific external commands
Closes #489: downtime error on cancel

### CODE REVIEW

File: alignak/external_command.py [hunk 1]
```diff
@@ -1623,8 +1781,8 @@ def disable_contactgroup_host_notifications(self, contactgroup):
         :type contactgroup: alignak.objects.contactgroup.Contactgroup
         :return: None
         """
-        for contact in contactgroup:
-            self.disable_contact_host_notifications(contact)
+        for contact_id in contactgroup.get_contacts():
```
Comment (ddurieux) [inline alignak/external_command.py]: Rename in `contact_uuid`

File: alignak/external_command.py [hunk 2]
```diff
@@ -1636,8 +1794,8 @@ def disable_contactgroup_svc_notifications(self, contactgroup):
         :type contactgroup: alignak.objects.contactgroup.Contactgroup
         :return: None
         """
-        for contact in contactgroup:
-            self.disable_contact_svc_notifications(contact)
+        for contact_id in contactgroup.get_contacts():
```
Comment (ddurieux) [inline alignak/external_command.py]: same => `contact_uuid`

File: alignak/external_command.py [hunk 3]
```diff
@@ -1733,8 +1876,9 @@ def disable_hostgroup_host_checks(self, hostgroup):
         :type hostgroup: alignak.objects.hostgroup.Hostgroup
         :return: None
         """
-        for host in hostgroup:
-            self.disable_host_check(host)
+        for host_id in hostgroup.get_hosts():
```
Comment (ddurieux) [inline alignak/external_command.py]: host_uuid

Comment (ddurieux) [inline alignak/external_command.py]: See all usage of host_id in end of file

File: alignak/objects/contact.py [hunk 4]
```diff
@@ -202,11 +238,13 @@ def want_service_notification(self, notifways, timeperiods, downtimes,
         if not self.service_notifications_enabled:
             return False
 
-        # If we are in downtime, we do nto want notification
+        # If we are in downtime, we do not want notification
         for downtime_id in self.downtimes:
             downtime = downtimes[downtime_id]
             if downtime.is_in_effect:
+                self.in_scheduled_downtime = True
```
Comment (ddurieux) [inline alignak/objects/contact.py]: Not sure it's the best place, it's better to define it when add downtime / delete a downtime on an item (create a function if not exist)

File: alignak/external_command.py (hunk 3 above)
Comment (mohierf) [inline alignak/external_command.py]: It is only an internal variable name ...

File: alignak/objects/contact.py (hunk 4 above)
Comment (mohierf) [inline alignak/objects/contact.py]: Hmmm ... why not, but this needs some more tests that cannot be made with unit tests :/P Let's leave this as is ... and we will see.

**Comment by mohierf** [comment]
Please merge as is because we need more tests for some of the features :)

**Pull request merged**